drivers/interrupt_controller/intc_gic.c

```diff
@@ -11,19 +11,80 @@
  *       GICv3 interface is not supported.
  */
 
-#include <device.h>
 #include <sw_isr_table.h>
-#include <irq_nextlevel.h>
 #include <dt-bindings/interrupt-controller/arm-gic.h>
 #include <drivers/interrupt_controller/gic.h>
 
 #if CONFIG_GIC_VER >= 3
 #error "GICv3 and above are not supported"
 #endif
 
-struct gic_ictl_config {
-	u32_t isr_table_offset;
-};
+void arm_gic_irq_enable(unsigned int irq)
+{
+	int int_grp, int_off;
+
+	int_grp = irq / 32;
+	int_off = irq % 32;
+
+	sys_write32((1 << int_off), (GICD_ISENABLERn + int_grp * 4));
+}
+
+void arm_gic_irq_disable(unsigned int irq)
+{
+	int int_grp, int_off;
+
+	int_grp = irq / 32;
+	int_off = irq % 32;
+
+	sys_write32((1 << int_off), (GICD_ICENABLERn + int_grp * 4));
+}
+
+int arm_gic_irq_is_enabled(unsigned int irq)
+{
+	int int_grp, int_off;
+	unsigned int enabler;
+
+	int_grp = irq / 32;
+	int_off = irq % 32;
+
+	enabler = sys_read32(GICD_ICENABLERn + int_grp * 4);
```

Comment:
This should be `GICD_ISENABLERn`

Reply:
As far as I can tell, read operations to the Interrupt Set-Enable Registers (`GICD_ISENABLERn`) and Interrupt Clear-Enable Registers (`GICD_ICENABLERn`) are equivalent (i.e. both registers read 1 if an interrupt is enabled and 0 if disabled).

![image](https://user-images.githubusercontent.com/11345418/74292844-70aeaa80-4d7c-11ea-943d-193f36c5a91a.png)

![image](https://user-images.githubusercontent.com/11345418/74292897-8c19b580-4d7c-11ea-865a-48d34a0ad38c.png)

Reply:
Nonetheless, I will change it to use the `GICD_ISENABLERn` registers since that could potentially make more intuitive sense.